packages/mix/lib/src/variants/context_variant_util/on_brightness_util.dart

```diff
@@ -21,7 +21,7 @@ class OnBrightnessVariant extends MediaQueryContextVariant {
   /// [brightness] associated with this variant, and `false` otherwise.
   @override
   bool when(BuildContext context) {
-    return Theme.of(context).brightness == brightness;
+    return MediaQuery.of(context).platformBrightness == brightness;
```

Comment:
Can we use the InheritedModel method of MediaQuery?
`MediaQuery.platformBrightnessOf(context) == brightness;`

Reply:
The main issue is the tests are focused on the brightness that attached to the apps theme. MediaQuery pulls from the device. Hence the tests failing. 

The goal is for $on.dark to work for Cupertino apps. Not 100% sure this ( MediaQuery ) is the right path.

Reply:
@doug-orchard The `platformBrightnessOf` will not change the behavior of your PR; it will only watch the `platformBrightness` value instead of any `MediaQuery` change.

Reply:
Okay! That sounds great.

Time and knowledge of Mix is holding me back to delve deeper that this change required. So, @leoafarias, feel free to take over.